pilot/pkg/bootstrap/server.go

```diff
@@ -291,6 +300,38 @@ func (s *Server) initMonitor(args *PilotArgs) error {
 	return nil
 }
 
+func (s *Server) initMeshConfigProtocolClient(args *PilotArgs) error {
+	serverAddr := args.MCPServerAddr
+	conn, err := grpc.Dial(serverAddr, grpc.WithInsecure())
+	if err != nil {
+		log.Infof("Unable connecting to MCP Server: %v\n", err)
+		return err
+	}
+
+	cl := mcp.NewAggregatedMeshConfigServiceClient(conn)
+	//TODO: make these supported types configurable
+	// currently mcp server has to also load all the supported types too
+	clientNodeID := "some-id"
+	supportedTypes := make([]string, len(ConfigDescriptor))
+	for i, model := range ConfigDescriptor {
+		supportedTypes[i] = model.MessageName
+	}
+	log.Infof("InitMeshConfigProtocolClient supported types: %+v\n", supportedTypes)
+	u := coredatamodel.NewUpdater(s.istioConfigStore, ConfigDescriptor)
```

Comment:
It looks like MCP is populating the existing `istioConfigStore` instead of being its own configStore. How are config merge conflicts handled? 

Should the the MCP client be initialized in `initConfigControllers` along side the file and k8s based controllers? e.g.

```golang
if args.MCPServerAddr != nil {
        // use MCP 
} else if args.Config.Controller != nil {
        s.configController = args.Config.Controller
} else if args.Config.FileDir != "" {
        // use file monitor + copilot    
} 
```

Reply:
>> It looks like MCP is populating the existing istioConfigStore instead of being its own configStore. How are config merge conflicts handled?

If Pilot MCP is enabled, have it populate the config store directly and expect only MCP config to be sent over the wire.

>> Should the the MCP client be initialized in initConfigControllers along side the file and k8s based controllers?

Yeahhh. Do we want to allow users to send both types of configs at once? Do we want to put some work into ensuring a decent debugging path in that use case? We could definitely make this change, but I wanted to confirm with you first whether we plan on supporting both at the same time.

Reply:
Use a different store. The k8s will be kept around for some time, but it would be best to not touch that code and to keep everything separate.